### PR TITLE
Remove language doc ztests

### PR DESCRIPTION
**Pull request by philrz**
**tl;dr** - These tests are no longer needed and can be removed.

**Details** - These were originally added in #2901 after several incidents where non-mdtest-protected examples in the docs kept breaking and we had to catch/fix them manually. However, in the time since they were added, the examples they were protecting disappeared during a docs overhaul that brought language intro out of the top-level `docs/language/README.md` and into `docs/language/overview.md` with most of the content/examples being redone at that time. I also see that in the ~1.5 years since the ztests were added, per the commit log, nobody has ever had to make changes to them, so it seems we've gotten better about not breaking the fundamentals. If we want ideal coverage, perhaps at some point we can take on #2908.

### CODE REVIEW

**Comment by nwt** [comment]
@philrz: Before merging this, let's verify that it doesn't reduce coverage.

**Review by nwt — APPROVED** [review]

**Comment by nwt** [comment]
@philrz: Let's merge this.
